integrations/optimum/src/haystack_integrations/components/embedders/backends/optimum_backend.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any, ClassVar, Dict, List, Optional, Union
+
+import numpy as np
+import torch
+from haystack.utils.auth import Secret
+from haystack_integrations.components.embedders.pooling import Pooling, PoolingMode
+from optimum.onnxruntime import ORTModelForFeatureExtraction
+from tqdm import tqdm
+from transformers import AutoTokenizer
+
+
+class _OptimumEmbeddingBackendFactory:
```

Comment:
Currently, this class is just being used to reuse instances of the model when its loaded. Since the ONNX runtime backend management is being handled by `ORTModelForFeatureExtraction`, I guess we don't need this abstraction.

I will simplify the implementation so that we directly use `OptimumEmbeddingBackend` for both the embedders.

Reply:
Yeah, let's directly use the backend class for now. Also, the cache key was under-specified, leading to subtle conflicts when the same model was used with different model args.